### DISCUSSION TITLE
HOW TO: Use Ansible Semaphore with "Roles"

### DISCUSSION THREAD

**Discussion by Ashkaan** [semaphoreui/semaphore#1455] [Q&A] [answered]
I have a playbook that has this in it:

`  roles:
    - name: install oh my zsh
      role: gantsign.oh-my-zsh
      users:
        - username: ashkaan
          oh_my_zsh:
            theme: agnoster
            plugins:
              - git
              - zsh-syntax-highlighting
              - zsh-autosuggestions
              - zsh-history-substring-search
              - auto-notify`

That fails everytime because, of course, my Semaphore docker container does not have the necessary role installed.  How do I install it?  How do I keep it persistent?

Thanks so much!

**Comment by craigarms** [comment, marked as answer]
You need as to root of your repo a roles directory with a requirements.yml file compliant with Ansible's requirement [file syntax](https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#installing-multiple-roles-from-a-file)

Each time you run your task it will read this file and attempt to install the roles specified.

**Comment by Ashkaan** [comment]
Yes!   Thank you so much.  That's it.
